.travis.yml

```diff
@@ -9,10 +10,38 @@ env:
   - secure: MvJDJRn23aWIVtcZVRg80nFYrk9wDlLAI6AZWgbCHfoVSu/PmoOnTVALxQJ1qVm2ywUsSvn0Oc08slGNfweRf4bhfeq9NFx5zxB3KNwxMPH66jU3Dri02tHXECxH9/cwyD5g8Pkwh2oVIWXbjgiNYQgFzcDLyhgToIBSCGMIkNdvAoK9lRVmKScwPQp5LqeolFbw8FcTcN9ozRsM7RewQXloo/FmUKMpBc5qRaZgEdnj16qYqCwXgQIrDmt0fixhdyvWBHWeTssO7jvcSuhCkjgbO0OinKID+299Bekm6fIoISFLptTHfJHdCRA7sBPv3vqP3VuA2h71PDReIbDg6pjymAwWACXy9Dl/dLnx5CbkGawym8ZzpLxXUBLqYcrEy6fsigFOH2BWkFAKHPEEjHS2cwHfHmoftTwhN2U4MkK8SYL5a2w0y5EjaETG20m7/SIyWq094SVhpRJhDjEb9oW61LtcH+CODNJzsHf3F8AZH1Igew+zutJlkC30fEKJ61yvS69LZmhSNCLpGAosmZWyR41t0S2XYHPb6M8VDJrezl2E99UsHrJkepfjhP9whbZHstbToh/m8xhKHb0Ple6gRYZ7uqOiPbOc9FjN7pH/sNDn3n9fymS8qXddJK2qd/rAjJ9St+4mzq/5heEgP/gyjQwwOcFul32jC7ds7Tg=
   - secure: V0zGqKSrp6faxVVQd8nW16+5XgeiyC1ZXojkCN4V91GDh7FhUfkM63liEpQhGOnRnQ9Z+ibmCSiTvS8kSCjkuME7rKC7jfmvEnsIQhGGk7GALoZfjP3Ox78czGFkZ8C3e6Y6rBStVoZDXXj6QpxWQxvYbFsf295g8QXjOliRfUchGz/H++OMbOcM1vNWQCfGNsRQ+0xWDYbk/z2V1XVGU1yxuCPDozWiwH2JjU4i8zzAhaqFua7ikfmneDhrTqd9BpR5cTfww66Tvb8KfqmQRrXoNelYMGoOqz/raVxP2+WtJr1sIhT/q6/rX9Bizl6+S5Pn7h9YES5eNOmIkIHwDgiP4SUwnT4PwuJw2PyAVCzn0WHceLGevHs8Wid82ONlXfON5gNJSpVUzcy7cym0LPbA60wL0MxjICmWXJklgdYqIruYZX9ijscwH4rWia4Q83n3BcSyzHr4g37Fu0f0W5cqcinNRV+dtQ0LdZA9yFH2qDMV1yngrUmDcnlMvMKnRwzDITzauDM6TmAXfphkNOTR32c1g4Rl84EFAqGtYI6tQoAAUm3ZEEMJzcL8gviCWqzQx3UkRq4UP2C7yF21p/2RqrhdAkeZU55J6OhymCdFxEjJswNlUj0y1u+mSw0ARmAO4ZtZ7XCQ1G3sBcJz8D4d4gx1Hc/HzHPERn8oZpE=
-after_success:
-- source ./build/travis-utils.sh
-- if is_travis_push_env; then docker login -u $DOCKER_USER -p $DOCKER_PASSWORD; fi
-- if is_travis_push_env; then make image-quick; fi
-- if is_travis_push_env; then make push; fi
-- if is_travis_push_env; then make tag-latest; fi
-- if is_travis_push_env; then make push-latest; fi
+jobs:
+  include:
+    - stage: "build, test, deploy"
+      script: make
+      deploy:
+        - provider: script
+          skip_cleanup: true
+          script: make deploy-travis
+          on:
+            repo: open-policy-agent/opa-istio-plugin
+            branch: master
+            tags: false
+    - stage: e2e
+      if: (NOT type IN (pull_request)) AND (branch = master)
+      before_script:
+      - KIND_VERSION=0.4.0
+      - ISTIO_VERSION=1.2.5
+      # Download and install kind
+      - curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output kind && chmod +x kind && sudo mv kind /usr/local/bin/
+      # Create kind cluster
+      - |
+        if [ $(kind get clusters) ]; then
+          export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
```

Comment:
If I'm interpreting this correctly, this lets it re-use clusters? Do we want to allow that for testing? I ask because I didn't see anything in the tests or script to cleanup afterwards.

Reply:
Added:
```yaml
after_script:
      - kind delete cluster
```